src/server/api/services/EventProcessor.ts

```diff
@@ -124,19 +121,23 @@ export class ObservationProcessor implements EventProcessor {
       metadata ?? undefined,
     );
 
-    const prompts =
+    const prompt =
       "promptName" in this.event.body &&
       typeof this.event.body.promptName === "string" &&
       "promptVersion" in this.event.body &&
       typeof this.event.body.promptVersion === "number"
-        ? await prisma.prompt.findMany({
+        ? await prisma.prompt.findUnique({
             where: {
-              projectId: apiScope.projectId,
-              name: this.event.body.promptName,
-              version: this.event.body.promptVersion,
+              projectId_name_version: {
+                projectId: apiScope.projectId,
+                name: this.event.body.promptName,
+                version: this.event.body.promptVersion,
+              },
             },
           })
         : undefined;
+    if (prompt === null)
+      console.warn("Prompt not found for observation", this.event.body);
```

Comment:
wont this be quite noisy?

Reply:
added comment to clarify that this is only logged in error cases